### PR TITLE
Rebuild the crate and add kvs support

### PR DESCRIPTION
**Pull request by scotow**

### CODE REVIEW

File: src/builder.rs [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+    }
+
+    /// Enable or disable automatic null character (`\0`) appending at the end
+    /// of every record. This may be required by some backends.
+    pub fn null_character(mut self, enabled: bool) -> Self {
+        self.null_character = enabled;
+        self
+    }
+
+    /// Enable or disable automatic appending type suffix to additional fields according to this [documentation](https://help.ovhcloud.com/csm/en-logs-data-platform-field-naming-conventions?id=kb_article_view&sysparm_article=KB0055662).
```
Comment (guillax) [inline src/builder.rs]: Should this comment wrap ?

File: src/builder.rs [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            additional_fields: flatten(self.additional_fields, Some("_"), "_", self.type_suffix)
+                .into_iter()
+                .chain(self.raw_additional_fields)
+                .collect(),
+        })
+    }
+}
+
+impl Default for Builder {
+    /// Crate a default builder that will log every record to stderr, with no
```
Comment (guillax) [inline src/builder.rs]:
```suggestion
    /// Creates a default builder that will log every record to `stderr`, with no
```

File: src/logger.rs [hunk 3]
```diff
@@ -2,31 +2,313 @@
...
+use env_filter::Filter;
+use log::{LevelFilter, Log, Metadata, Record};
+use native_tls::{TlsConnector, TlsStream};
+use serde_json::{Map, Value};
 
-pub(crate) struct GelfLogger {
-    level: Level,
+use crate::{error::Error, record::GelfRecord, Builder};
+
+/// A logger what will format and forward any [`Record`] to the set-up target.
```
Comment (guillax) [inline src/logger.rs]:
```suggestion
/// A logger that will format and forward any [`Record`] to the set-up target.
```

**Review by guillax — APPROVED** [review]
Looks ok to me except few typos

File: src/builder.rs (hunk 1 above)
Comment (scotow) [reply src/builder.rs]: I will do it manually, cargo fmt doesn't do it by itself.

**Pull request merged**